### ISSUE TITLE
Error when attempting to import KALQ layout

### ISSUE THREAD

**Issue by Ferk** [Lixquid/app-unexpected-keyboard-layout-editor#8]
I noticed I get an error when loading the XML below (taken from [the KALQ layout](https://github.com/JackDotJS/Unexpected-Keyboard/blob/feat-kalq-layout/res/xml/kalq.xml) that was posted in https://github.com/Julow/Unexpected-Keyboard/issues/308#issuecomment-1595762691).
```
<?xml version="1.0" encoding="utf-8"?>
<keyboard bottom_row="false">
  <row height="0.7913">
    <key key0="m" key2="("/>
    <key key0="b" key2=")"/>
    <key key0="w" key2="1"/>
    <key key0="h" key2="2"/>
    <key shift="0.5" key0="g" key1="3"/>
    <key key0="t" key1="+"/>
    <key key0="o" key1="&amp;"/>
    <key key0="j" key1="\?"/>
  </row>
  <row height="0.7913"> 
    <key key0="p" key2="\@"/>
    <key key0="space" key2="\#"/>
    <key key0="x" key2="4"/>
    <key key0="c" key2="5"/>
    <key shift="0.5" key0="i" key1="6" key3="="/>
    <key key0="e" key1="-" key3="\\"/>
    <key key0="space" key1="_" key3=";"/>
    <key key0="u" key1="!" key3=":"/>
  </row>
  <row height="0.7913">
    <key key0="r" key2=","/>
    <key key0="y" key2="%"/>
    <key key0="s" key2="7"/>
    <key key0="z" key2="8"/>
    <key shift="0.5" key0="k" key1="9" key3="["/>
    <key key0="a" key1="*" key3="]"/>
    <key key0="l" key1="&quot;"/>
    <key key0="q" key1="'"/>
  </row>
  <row height="0.7913">
    <key key0="d" key2="." key4="~"/>
    <key key0="n" key2="$" key4="|"/>
    <key key0="f" key2="/" key4="&lt;"/>
    <key key0="v" key2="0" key4="&gt;"/>
    <key shift="0.5" width="1.3333" key0="backspace" key2="delete"/>
    <key width="1.3333" key7="up" key6="right" key5="left" key8="down"/>
    <key width="1.3333" key0="enter" key2="action"/>
  </row>
  <row height="0.7913">
    <key width="2" key0="shift" key2="loc capslock" key4="loc meta"/>
    <key width="2" key0="ctrl" key1="change_method" key2="config" key3="loc alt" key4="fn"/>
    <key shift="0.5" width="2" key0="switch_numeric" key1="switch_emoji"/>
    <key width="2" key0="shift" key5="left" key6="right" slider="true"/>
  </row>
</keyboard>
```

Error from the javascript console below:
```
Uncaught (in promise) TypeError: e is undefined
    wt KeyLegend.tsx:69
    Preact 32
    onClick NewCard.tsx:90
    Preact 30
    <anonymous> index.tsx:4
[KeyLegend.tsx:69:8](https://unexpected-keyboard-layout-editor.lixquid.com/src/components/KeyLegend.tsx)
```

**Comment by Lixquid** [comment]
Hey, thanks for the report! Looks like it's from the 6th key in the 4th row; it has no `key0` attribute, which I had assumed was mandatory; it's managed to slip past all of my type checks.

As an interim you can set it to an empty value; I'll change the tool to allow optional `key0` attributes and try to add some more user-friendly error reporting on XML import.

**Comment by Lixquid** [comment]
Fixed in afb7acd
